test/functional/mempool_updatefromblock.py

```diff
@@ -38,86 +35,52 @@ def transaction_graph_test(self, size, n_tx_to_mine=None, start_input_txid='', e
-            tx_size.append(self.nodes[0].getrawmempool(True)[tx_id[-1]]['vsize'])
-
-            if tx_count in n_tx_to_mine:
-                # The created transactions are mined into blocks by batches.
-                self.log.info('The batch of {} transactions has been accepted into the mempool.'.format(len(self.nodes[0].getrawmempool())))
-                block_hash = self.nodes[0].generate(1)[0]
-                if not first_block_hash:
-                    first_block_hash = block_hash
-                assert_equal(len(self.nodes[0].getrawmempool()), 0)
+        self.wallet.generate(1)
+        node.generate(size)
+        txs = []
+        block_ids = []
+        tx_size = []
+        for i in range(size):
+            txs.append(self.wallet.send_self_transfer(from_node=node))
+            tx_size.append(96)
+            if (i+1) % n_tx_to_mine == 0:
+                self.log.info('The batch of {} transactions has been accepted into the mempool.'
+                .format(len(node.getrawmempool())))
+                block_ids.append(self.wallet.generate(1)[0])
+                assert_equal(len(node.getrawmempool()), 0)
                 self.log.info('All of the transactions from the current batch have been mined into a block.')
-            elif tx_count == size:
-                # At the end all of the mined blocks are invalidated, and all of the created
-                # transactions should be re-added from disconnected blocks to the mempool.
-                self.log.info('The last batch of {} transactions has been accepted into the mempool.'.format(len(self.nodes[0].getrawmempool())))
-                start = time.time()
-                self.nodes[0].invalidateblock(first_block_hash)
-                end = time.time()
-                assert_equal(len(self.nodes[0].getrawmempool()), size)
-                self.log.info('All of the recently mined transactions have been re-added into the mempool in {} seconds.'.format(end - start))
+                self.log.info('{}'.format(block_ids))
+                self.log.info('Block height: {}'.format(math.ceil(i / n_tx_to_mine)))
+
+        self.log.info('Mempool size pre-invalidation: {}'.format(len(node.getrawmempool())))
+        assert_equal(len(node.getrawmempool()), 0)
+        # Invalidate the first block to send the transactions back to the mempool.
+        start = time.time()
+        node.invalidateblock(block_ids[0])
+        end = time.time()
+        self.log.info('Mempool size post-invalidation: {}'.format(len(node.getrawmempool())))
+        assert_equal(len(self.nodes[0].getrawmempool()), size)
+        self.log.info('All of the recently mined transactions have been re-added into the mempool in {} seconds.'
+        .format(end - start))
 
         self.log.info('Checking descendants/ancestors properties of all of the in-mempool transactions...')
-        for k, tx in enumerate(tx_id):
+        k = 0
+        for tx in txs:
```

Comment:
I'm curious if there's any advantage to incrementing k manually rather than using `enumerate`? Otherwise `enumerate` seems more concise.
```suggestion
        for k, tx in enumerate(txs):
```

Reply:
I checked and there is no significant difference in test execution time. Using enumerate ac8b08b